### PR TITLE
Log errors in multiplayer and the client for pubsub errors

### PR DESCRIPTION
**Pull request by ddimaria**
- [x] Convert Redis connection warnings in Multiplayer to errors
- [x] Send multiplayer errors in the frontend to Sentry

Closes: https://github.com/quadratichq/quadratic/issues/1269

Console errors now for multiplayer error messages sent back to the client:
Screenshot 2024-04-08 at 3 22 58 PM

### CODE REVIEW

File: quadratic-client/src/multiplayer/multiplayer.ts [hunk 1]
```diff
@@ -603,7 +604,15 @@ export class Multiplayer {
     } else if (type === 'CurrentTransaction') {
       this.receiveCurrentTransaction(data);
     } else if (type === 'Error') {
-      console.warn(`[Multiplayer] Error`, data.error);
+      if (data.error_level === 'Error') {
+        Sentry.captureException({
+          message: `Error response from the multiplayer server: ${data.error}`,
```
Comment (davidkircos) [inline quadratic-client/src/multiplayer/multiplayer.ts:609]: In the logs we get this error all the time from Multiplayer. Will this also trigger Sentry now?
CleanShot 2024-04-08 at 23 15 11@2x

File: quadratic-multiplayer/src/server.rs [hunk 2]
```diff
@@ -197,10 +200,19 @@ async fn handle_socket(
             Ok(ControlFlow::Continue(_)) => {}
             Ok(ControlFlow::Break(_)) => break,
             Err(error) => {
-                tracing::warn!("Error processing message: {:?}", &error);
+                let mut error_level = ErrorLevel::Warning;
+
+                match error {
+                    MpError::PubSub(_) => {
+                        tracing::error!("Error processing message: {:?}", &error);
```
Comment (davidkircos) [inline quadratic-multiplayer/src/server.rs:207]: Should we add a test? Code Cov is complaining

File: quadratic-client/src/multiplayer/multiplayer.ts (hunk 1 above)
Comment (ddimaria) [reply quadratic-client/src/multiplayer/multiplayer.ts:609]: If you look at the Rust code, we only send an error_level of `Error` for PubSub errors, not the other errors in the logs.  Are there PubSub errors that you think should be warnings?

File: quadratic-multiplayer/src/server.rs (hunk 2 above)
Comment (ddimaria) [reply quadratic-multiplayer/src/server.rs:207]: I may be able to add a test for this unique case (assuming I can simulate a PubSub error in an integration test (tests on server.rs need to be integration tests).  I'll report back and timebox the effort to 2 hours.

Comment (ddimaria) [reply quadratic-multiplayer/src/server.rs]: I refactored this logic to occur outside of this function for better isolation.

File: quadratic-client/src/multiplayer/multiplayer.ts (hunk 1 above)
Comment (davidkircos) [reply quadratic-client/src/multiplayer/multiplayer.ts:609]: Makes sense

**Review by davidkircos — APPROVED** [review]

**Pull request merged**